draft-ietf-gnap-core-protocol.md

```diff
@@ -5365,3 +5365,693 @@ justification for the data type representing the same basic kind of thing it ext
+        }
+      },
+      "allOf": [
+        {
+          "if": {
+            "properties": {
+              "subject_type": {
+                "const": "account"
+              }
+            }
+          },
+          "then": {
+            "properties": {
+              "uri": {
+                "type": "string"
+              }
+            },
+            "required": [
+              "uri"
+            ]
+          }
+        },
+        {
+          "if": {
+            "properties": {
+              "subject_type": {
+                "const": "email"
+              }
+            }
+          },
+          "then": {
+            "properties": {
+              "email": {
+                "type": "string"
+              }
+            },
+            "required": [
+              "email"
+            ]
+          }
+        },
+        {
+          "if": {
+            "properties": {
+              "subject_type": {
+                "const": "phone-number"
+              }
+            }
+          },
+          "then": {
+            "properties": {
+              "phone_number": {
+                "type": "string"
+              }
+            },
+            "required": [
+              "phone_number"
+            ]
+          }
+        },
+        {
+          "if": {
+            "properties": {
+              "subject_type": {
+                "const": "iss-sub"
+              }
+            }
+          },
+          "then": {
+            "properties": {
+              "iss": {
+                "type": "string"
+              },
+              "sub": {
+                "type": "string"
+              }
+            },
+            "required": [
+              "uri",
+              "sub"
+            ]
+          }
+        }
+      ]
+    }
+  },
+  "type": "object",
+  "properties": {
+    "resources": {
+      "type": [
+        "array",
+        "object"
+      ],
+      "items": {
+        "$ref": "#/definitions/resource"
+      },
+      "additionalProperties": {/* if it's an object for multiple tokens, each field's value is a resource object*/
+        "type": [
+          "array"
+        ],
+        "items": {
+          "$ref": "#/definitions/resource"
+        }
+      }
+    },
+    "subject": {
+      "type": "object",
+      "properties": {
+        "sub_ids": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/subject_id_types"
+          }
+        },
+        "assertions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/assertion_types"
+          }
+        }
+      }
+    },
+    "client": {
+      "oneOf": [
+        {
+          "type": "object",
+          "properties": {
+            "instance_id": {
+              "type": "string"
+            }
+          },
+          "additionalProperties": false
+        },
+        {
+          "type": [
+            "object",
+            "string"
```

Comment:
This string here is meant to represent a reference, right? If so I would instead point to a separate definition of a reference, both to make it clearer and to allow us to change the reference format in the future, in one place.

Reply:
We can do that, but right now it's defined in the protocol as a string and so I left it as such instead of trying to abstract it ahead of time.